compiler/wasm/src/types/noir_artifact.ts

```diff
@@ -0,0 +1,200 @@
+import { Abi, AbiParameter } from '@noir-lang/noirc_abi';
+
+/**
+ * A contract event.
+ */
+export interface EventAbi {
+  /**
+   * The event name.
+   */
+  name: string;
+  /**
+   * Fully qualified name of the event.
+   */
+  path: string;
+  /**
+   * The fields of the event.
+   */
+  fields: AbiParameter[];
+}
+
+/** The Noir function types. */
+export type NoirFunctionType = 'Open' | 'Secret' | 'Unconstrained';
+
+/**
+ * The compilation result of an Noir function.
+ */
+export interface NoirFunctionEntry {
+  /** The name of the function. */
+  name: string;
+  /** The type of the function. */
+  function_type: NoirFunctionType;
+  /** Whether the function is internal. */
+  is_internal: boolean;
+  /** The ABI of the function. */
+  abi: Abi;
+  /** The bytecode of the function in base64. */
+  bytecode: string;
+  /** The proving key. */
+  proving_key: string;
+  /** The verification key. */
+  verification_key: string;
```

Comment:
These fields do not exist on a compiled contract's functions

Reply:
Mmm, this was ported from `aztec-packages` as-is...is that a bug on the other side?

Reply:
I think generally we can't port stuff from `aztec-packages` and assume that it's going to work. They had an independent compiler implementation for quite some time so picked up quite a few quirks, especially around build artifacts.